### PR TITLE
Add log_links to GetTaskResponse

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue
https://github.com/flyteorg/flyte/issues/3936

## Why are the changes needed?
To show the log link on FlyteConsole

## What changes were proposed in this pull request?
- Add log_links to GetTaskResponse
- Show the log link in local execution
- BigQuery returns a log link

### Setup process
1. Enable BigQuery agent
2. Run a BigQuery Task
```python
import pandas as pd
import pyarrow as pa
from flytekit import StructuredDataset, kwtypes, task, workflow
from flytekitplugins.bigquery import BigQueryConfig, BigQueryTask
from typing_extensions import Annotated

MyDataset = Annotated[StructuredDataset, kwtypes(name=str)]


@task
def create_bq_table() -> StructuredDataset:
    df = pd.DataFrame(data={"name": ["Alice", "bob"], "age": [5, 6]})
    return StructuredDataset(
        dataframe=df, uri="bq://dogfood-gcp-dataplane:dataset.flyte_table3"
    )


bigquery_task_templatized_query = BigQueryTask(
    name="bigquery",
    inputs=kwtypes(version=int),
    output_structured_dataset_type=MyDataset,
    task_config=BigQueryConfig(ProjectID="dogfood-gcp-dataplane"),
    query_template="SELECT * from dataset.flyte_table3;",  # type: ignore
)


@task
def convert_bq_table_to_arrow_table(sd: MyDataset) -> pa.Table:
    t = sd.open(pa.Table).all()
    print(t)
    return t


@workflow
def wf(version: int = 10) -> pa.Table:
    create_bq_table()
    sd = bigquery_task_templatized_query(version=version)
    return convert_bq_table_to_arrow_table(sd=sd)


if __name__ == "__main__":
    wf()

```

### Screenshots
image

image

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: flytekit/extend/backend/base_agent.py [hunk 1]
```diff
@@ -215,7 +216,9 @@ async def _get(self, resource_meta: bytes) -> GetTaskResponse:
             else:
                 res = self._agent.get(grpc_ctx, resource_meta)
             state = res.resource.state
-            logger.info(f"Task state: {state}, State message: {res.resource.message}")
+            print(f"{self._entity.name} task state: {State.Name(int(state))}, message: {res.resource.message or None}")
+            for link in res.log_links:
+                print(f"{link.name}: {link.uri}")
```
Comment (eapolinario) [inline flytekit/extend/backend/base_agent.py:221]: should we use the logger here?

File: flytekit/extend/backend/base_agent.py [hunk 2]
```diff
@@ -215,7 +216,9 @@ async def _get(self, resource_meta: bytes) -> GetTaskResponse:
             else:
                 res = self._agent.get(grpc_ctx, resource_meta)
             state = res.resource.state
-            logger.info(f"Task state: {state}, State message: {res.resource.message}")
+            print(f"{self._entity.name} task state: {State.Name(int(state))}, message: {res.resource.message or None}")
```
Comment (kumare3) [inline flytekit/extend/backend/base_agent.py:219]: here too

Comment (kumare3) [reply flytekit/extend/backend/base_agent.py:219]: you probably want this in the agent executor mixin?

File: plugins/flytekit-bigquery/flytekitplugins/bigquery/agent.py [hunk 3]
```diff
@@ -79,12 +80,20 @@ def create(
     def get(self, context: grpc.ServicerContext, resource_meta: bytes) -> GetTaskResponse:
         client = bigquery.Client()
         metadata = Metadata(**json.loads(resource_meta.decode("utf-8")))
+        log_links = [
+            TaskLog(
```
Comment (kumare3) [inline plugins/flytekit-bigquery/flytekitplugins/bigquery/agent.py:84]: missing ttl?

**Comment by kumare3** [comment]
lgtm, couple changes

File: plugins/flytekit-bigquery/flytekitplugins/bigquery/agent.py (hunk 3 above)
Comment (pingsutw) [reply plugins/flytekit-bigquery/flytekitplugins/bigquery/agent.py:84]: what default value should we set here

File: flytekit/extend/backend/base_agent.py (hunk 2 above)
Comment (pingsutw) [reply flytekit/extend/backend/base_agent.py]: it is in the mixin.

**Review by Future-Outlier — APPROVED** [review]
LGTM, I've tested it in this PR.
https://github.com/flyteorg/flyte/pull/4529

**Review by Future-Outlier — DISMISSED** [review]
need to add tests, LGTM!

**Review by Future-Outlier — APPROVED** [review]
+1

**Review by eapolinario — APPROVED** [review]

**Pull request merged**